streams/src/main/java/org/apache/kafka/streams/processor/internals/InitProcessorRecordContext.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals;
+
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.apache.kafka.common.utils.Time;
+
+import java.util.Objects;
+
+public class InitProcessorRecordContext extends ProcessorRecordContext {
+
+    private final Time time;
+
+    public InitProcessorRecordContext(final Time time) {
+        super(time.milliseconds(), -1, -1, null, new RecordHeaders());
```

Comment:
```suggestion
        super(-1L, -1, -1, null, new RecordHeaders());
```

Or maybe even better, add `private final static long NO_TIMESTAMP = -1L;` and pass `NO_TIMESTAMP` here, or use `ConsumerRecord.NO_TIMESTAMP` directly ?

Reply:
Better. Thanks!

Reply:
I also added `private static final long NO_OFFSET = -1;` and `private static final int NO_PARTITION = -1;`

Reply:
I was thinking about this more, as discussed on Zoom recently, we both tend to agree that not using a fixed value might be semantically better. However, as we use a fixed value inside `process()` and `punctuate()` callback, I am concerned about consistency.

So I would propose to use a fixed value for this PR. -- I am open to change it to a dynamic value, but for this case, we should also change it for `process()` and `punctuate()`, what would be a larger change (and unrelated to this bug fix).

Happy to file a Jira ticket about it, and you can pick it up if you want to.

Reply:
That sounds good! 

We can discuss the moving timestamp for the `process()` and `punctuate()` methods. Overhead is the main concern, but we can still discuss its value.

Reply:
I already updated this class to use a static ts for the init method. 👍